app/shared/actions/stake.js

```diff
@@ -1,53 +1,56 @@
-import * as validate from './validate';
+import { Decimal } from 'decimal.js';
+
 import * as types from './types';
 
-import { delegatebw } from './system/delegatebw';
-import { undelegatebw } from './system/undelegatebw';
+import { delegatebwParams } from './system/delegatebw';
+import { undelegatebwParams } from './system/undelegatebw';
 
-import {Decimal} from 'decimal.js';
+import * as AccountActions from './accounts';
+import eos from './helpers/eos';
 
-export function setStakeWithValidation(EOSbalance, account, netAmount, cpuAmount) {
-  return (dispatch: () => void) => {
-    const { nextStake, currentStake } = getNextAndCurrentStake(account, netAmount, cpuAmount);
+export function setStake(account, netAmount, cpuAmount) {
+  return (dispatch: () => void, getState) => {
+    const {
+      connection
+    } = getState();
 
-    if (dispatch(validate.validateStake(nextStake, currentStake))) {
-      const increaseInStake = {
-        netAmount: Math.max(0, (nextStake.netAmount - currentStake.netAmount)),
-        cpuAmount: Math.max(0, (nextStake.cpuAmount - currentStake.cpuAmount))
-      };
+    const {
+      increaseInStake,
+      decreaseInStake
+    } = getStakeChanges(account, netAmount, cpuAmount);
 
-      const decreaseInStake = {
-        netAmount: Math.max(0, (currentStake.netAmount - nextStake.netAmount)),
-        cpuAmount: Math.max(0, (currentStake.cpuAmount - nextStake.cpuAmount))
-      };
+    dispatch({ type: types.VALIDATE_STAKE_PENDING });
 
+    return eos(connection).transaction(tr => {
       if (increaseInStake.netAmount > 0 || increaseInStake.cpuAmount > 0) {
-        dispatch(delegatebw(
+        tr.delegatebw(delegatebwParams(
           account.account_name,
           account.account_name,
           increaseInStake.netAmount,
           increaseInStake.cpuAmount
         ));
       }
       if (decreaseInStake.netAmount > 0 || decreaseInStake.cpuAmount > 0) {
-        dispatch(undelegatebw(
+        tr.undelegatebw(undelegatebwParams(
           account.account_name,
           account.account_name,
           decreaseInStake.netAmount,
           decreaseInStake.cpuAmount
         ));
       }
-    }
-  };
-}
-
-export function setStakeConfirmingWithValidation(EOSbalance, account, netAmount, cpuAmount) {
-  return (dispatch: () => void) => {
-    const { nextStake, currentStake } = getNextAndCurrentStake(account, netAmount, cpuAmount);
+    }).then((tx) => {
+      dispatch(AccountActions.getAccount(account.account_name));
 
-    if (dispatch(validate.validateStake(nextStake, currentStake))) {
-      return dispatch({ type: types.VALIDATE_STAKE_CONFIRMING });
-    }
+      return dispatch({
+        payload: { tx },
+        type: types.VALIDATE_STAKE_SUCCESS
```

Comment:
This isn't a validate action - is this needed? This also applies `VALIDATE_STAKE_FAILURE` on the lines below and the `VALIDATE_STAKE_NULL` within `resetStakeForm()`.

Reply:
I could use a SYSTEM_STAKE_SUCCESS method instead and move the logic to a system/stake file. What do you think?

Reply:
Actually what do you think of plain STAKE_SUCCESS?

Reply:
Use the `SYSTEM` methods (I think we talked about this in slack, just restating here)